packages/cosmic-swingset/lib/ag-solo/vats/offer-compiler.js

```diff
@@ -0,0 +1,213 @@
+import { assert, details } from '@agoric/assert';
+
+export default ({
+  E,
+  zoe,
+  registrar,
+
+  collections: {
+    idToOfferDesc,
+    brandToMath,
+    issuerToIssuerNames,
+    issuerToBrand,
+    purseToIssuer,
+    petnameToPurse,
+  },
+}) => async (id, offerDesc, hooks = {}) => {
+  const {
+    instanceRegKey,
+    contractIssuerIndexToRole = [], // FIXME: Only for compatibility with Zoe pre-Roles
+    offerRulesTemplate,
+  } = offerDesc;
+
+  const roleIssuerNames = {};
+  function createRoleOfferRulesAndPurses(tmpl) {
+    const roleOfferRules = { exit: tmpl.exit };
+    const rolePurses = {};
+
+    const setPurseAmount = (roles, role, purse, extent = undefined) => {
+      const issuer = purseToIssuer.get(purse);
+      roleIssuerNames[role] = issuerToIssuerNames.get(issuer);
+      const brand = issuerToBrand.get(issuer);
+      let amount;
+      if (extent === undefined) {
+        amount = brandToMath.get(brand).getEmpty();
+      } else {
+        amount = { brand, extent };
```

Comment:
We could use the amountMath here for OO best practices since we already have access to AmountMath through the brand:
```suggestion
        amount = amountMath.make(extent);
```

Reply:
Good catch!  Done.